File: matiasmasso/Mat.NET/DTO/DTOBloggerPost.cs
using System;
using System.Collections.Generic;

namespace DTO
{
    public class DTOBloggerPost : DTOBaseGuid
    {
        public DTOBlogger Blogger { get; set; }
        public string Title { get; set; }
        public string Url { get; set; }
        public DateTime Fch { get; set; }
        public DTOLang Lang { get; set; }
        public DateTime HighlightFrom { get; set; }
        public DateTime HighlightTo { get; set; }
        public List<DTOProduct> Products { get; set; }

        public DTOBloggerPost() : base()
        {
            Products = new List<DTOProduct>();
        }

        public DTOBloggerPost(Guid oGuid) : base(oGuid)
        {
            Products = new List<DTOProduct>();
        }

        public static DTOBloggerPost Factory(DTOBlogger oBlogger = null/* TODO Change to default(_) if this is not a reference type */)
        {
            DTOBloggerPost retval = new DTOBloggerPost();
            {
                var withBlock = retval;
                withBlock.Blogger = oBlogger;
                withBlock.Fch = DTO.GlobalVariables.Today();
            }
            return retval;
        }

        public static string FullUrl(DTOBloggerPost oPost)
        {
            string retval = oPost.Url;
            if (!retval.StartsWith("http"))
                retval = "https://" + retval;
            return retval;
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOBloggerPost.vb
Public Class DTOBloggerPost
    Inherits DTOBaseGuid

    Property Blogger As DTOBlogger
    Property Title As String
    Property Url As String
    Property Fch As Date
    Property Lang As DTOLang
    Property HighlightFrom As Date
    Property HighlightTo As Date
    Property Products As List(Of DTOProduct)

    Public Sub New()
        MyBase.New()
        _Products = New List(Of DTOProduct)
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
        _Products = New List(Of DTOProduct)
    End Sub

    Shared Function Factory(Optional oBlogger As DTOBlogger = Nothing) As DTOBloggerPost
        Dim retval As New DTOBloggerPost
        With retval
            .Blogger = oBlogger
            .Fch = DateTime.Today
        End With
        Return retval
    End Function

    Shared Function FullUrl(oPost As DTOBloggerPost) As String
        Dim retval As String = oPost.Url
        If Not retval.StartsWith("http") Then
            retval = "https://" & retval
        End If
        Return retval
    End Function
End Class
